ruby/ql/lib/codeql/ruby/dataflow/internal/DataFlowPublic.qll

```diff
@@ -261,14 +260,14 @@ class LocalSourceNode extends Node {
    * See `TypeBackTracker` for more details about how to use this.
    */
   pragma[inline]
-  LocalSourceNode backtrack(TypeBackTracker t2, TypeBackTracker t) { t2 = t.step(result, this) }
+  LocalSourceNode backtrack(TypeBackTracker t2, TypeBackTracker t) { t = t2.step(result, this) }
```

Comment:
Is this change correct? Were the parameters of `backtrack` swapped? We might want to check that all the places where we use `backtrack` are still correct.

Reply:
Yes, it is because the shared library (which defines `step`) has the arguments reversed. I decided to keep the order as-is in `backtrack`.